tests/OmniSharp.Tests/FixUsingsFacts.cs

```diff
@@ -0,0 +1,342 @@
+}";
+            await AssertBufferContents(fileContents, fileContents);
+        }
+
+        [Fact]
+        public async Task FixUsings_AddsUsingForExtension()
+        {
+            const string fileContents = @"namespace nsA {
+    public static class StringExtension {
+        public static void Whatever(this string astring) {}
+    }
+}
+
+namespace OmniSharp {
+    public class class1 
+    {
+        public method1()
+        {
+            ""string"".Whatever();
+        }
+    }
+}";
+            string expectedFileContents = @"using nsA;
+
+namespace nsA {
+    public static class StringExtension {
+        public static void Whatever(this string astring) {}
+    }
+}
+
+namespace OmniSharp {
+    public class class1 
+    {
+        public method1()
+        {
+            ""string"".Whatever();
+        }
+    }
+}";
+
+            await AssertBufferContents(fileContents, expectedFileContents);
+        }
+
+
+        [Fact]
+        public async Task FixUsings_AddsUsingLinqMethodSyntax()
+        {
+            const string fileContents = @"namespace OmniSharp {
+    public class class1 
+    {
+        public void method1()
+        {
+            List<string> first = new List<string>();
+            var testing = first.Where(s => s == ""abc"");
+        }
+    }
+}";
+            string expectedFileContents = @"using System.Collections.Generic;
+using System.Linq;
+
+namespace OmniSharp {
+    public class class1 
+    {
+        public void method1()
+        {
+            List<string> first = new List<string>();
+            var testing = first.Where(s => s == ""abc"");
+        }
+    }
+}";
+
+            await AssertBufferContents(fileContents, expectedFileContents);
+        }
+
+        private async Task AssertBufferContents(string fileContents, string expectedFileContents)
+        {
+            var response = await RunFixUsings(fileContents);
+            Assert.Equal(response.Buffer, expectedFileContents);
+        }
+
+        private async Task AssertUnresolvedReferences(string fileContents, List<QuickFix> expectedUnresolved)
+        {
+            var response = await RunFixUsings(fileContents);
+            var qfList = response.AmbiguousResults.ToList();
+            Assert.Equal(qfList.Count(), expectedUnresolved.Count());
+            var i = 0;
+            foreach (var expectedQuickFix in expectedUnresolved)
+            {
+                Assert.Equal(qfList[i].Line, expectedQuickFix.Line);
+                Assert.Equal(qfList[i].Column, expectedQuickFix.Column);
+                Assert.Equal(qfList[i].FileName, expectedQuickFix.FileName);
+                Assert.Equal(qfList[i].Text, expectedQuickFix.Text);
+                i++;
+            }
+        }
+
+        private async Task<FixUsingsResponse> RunFixUsings(string fileContents)
+        {
+            var workspace = TestHelpers.CreateSimpleWorkspace(fileContents, fileName);
+            var controller = new OmnisharpController(workspace, null);
```

Comment:
You should pass in options here. Instead of null, use `new FakeOmniSharpOptions()`. You can modify the FakeOmniSharpOptions class to allow you to specify the desired newline character.